File: rb/features.rb
# MultifonClient SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MultifonClientFeatures
  def self.make_feature(name)
    case name
    when "base"
      MultifonClientBaseFeature.new
    when "test"
      MultifonClientTestFeature.new
    else
      MultifonClientBaseFeature.new
    end
  end
end

File: py/features.py
# MultifonClient SDK feature factory

from feature.base_feature import MultifonClientBaseFeature
from feature.test_feature import MultifonClientTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MultifonClientBaseFeature(),
        "test": lambda: MultifonClientTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
